build.gradle.kts

```diff
@@ -32,30 +32,44 @@ allprojects {
         mavenLocal()
     }
 
-    // Skip Javadoc generation for Java 1.8 as it breaks build
-    if (JavaVersion.current().isJava8Compatible) {
-        tasks.withType<Javadoc> {
-            options {
-                this as StandardJavadocDocletOptions
-                addStringOption("Xdoclint:none", "-quiet")
-            }
-        }
-    }
-
     if ((group as String).isNotEmpty() && name != "lint" && name != "internal") {
         configureAndroid()
         configureQuality()
 
-        val isLibrary = name == "library"
         if (Config.submodules.contains(name) || isLibrary) {
-            setupPublishing(isLibrary)
+            setupPublishing()
+        }
+    }
+}
+
+// Skip Javadoc generation for Java 1.8 as it breaks build
+if (JavaVersion.current().isJava8Compatible) {
+    tasks.withType<Javadoc> {
+        options {
+            this as StandardJavadocDocletOptions
+            addStringOption("Xdoclint:none", "-quiet")
         }
     }
 }
 
 val Project.configDir get() = "$rootDir/library/quality"
 val Project.reportsDir get() = "$buildDir/reports"
 
+/**
+ * Extension property to determine if a Project is the 'library' module
```

Comment:
`Extension property` sounds a little weird to me: it'd be like having javadoc that says `Method ...`. I think `Determines if a ...` sounds better. (Here and below)